uint/src/uint.rs

```diff
@@ -233,19 +232,12 @@ macro_rules! uint_overflowing_binop {
 					let (res1, overflow1) = ($fn)(me[i], you[i]);
 					let (res2, overflow2) = ($fn)(res1, carry);
 
-					unsafe {
-						// SAFETY: `i` is within bounds and `i * size_of::<u64>() < isize::MAX`
-						*ret_ptr.offset(i as _) = res2
-					}
+					ret[i] = res2;
 					carry = (overflow1 as u8 + overflow2 as u8) as u64;
 				} else {
 					let (res, overflow) = ($fn)(me[i], you[i]);
 
-					unsafe {
-						// SAFETY: `i` is within bounds and `i * size_of::<u64>() < isize::MAX`
-						*ret_ptr.offset(i as _) = res
-					}
-
+					ret[i] = res;
```

Comment:
did you also check https://doc.rust-lang.org/std/primitive.slice.html#method.get_unchecked_mut?